### PR TITLE
update FUIGoogleAuth for GoogleSignIn 6

### LINKED ISSUE #951: Add Google Auth to SPM
* Objective C or Swift: Swift
  * iOS version: iOS 14.1
  * Firebase SDK version: 7.11.1
  * FirebaseUI version: master branch
  * CocoaPods Version: Swift Package Manager
  
### Step 3: Describe the problem:

When using with FirebaseUI with SPM, FUIGoogleAuth cannot be found in the project. Since I can find it in the spm packages, I assume that something is not exported right.

Bildschirmfoto 2021-05-17 um 10 10 23

Bildschirmfoto 2021-05-17 um 10 02 21

#### Steps to reproduce:

  1. Add FirebaseUI with SPM
  2. Try to import and use FUIGoogleAuth
  
#### Observed Results:

  * Cannot find *FUIGoogleAuth' in scope
  
#### Expected Results:

  * Can find and use FUIGoogleAuth

### PR DESCRIPTION
**Pull request by morganchen12**
Fixes #951.

### CODE REVIEW

File: FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h [hunk 1]
```diff
@@ -85,6 +89,12 @@ __attribute__((deprecated("Instead use initWithAuthUI:")));
 - (instancetype)initWithScopes:(NSArray <NSString *> *)scopes
 __attribute__((deprecated("Instead use initWithAuthUI:permissions:"))) NS_DESIGNATED_INITIALIZER;
 
+/** @fn initWithAuthUI:scopes:
+    @brief Requests the scopes in the `scopes` array.
+ */
+- (void)requestScopesWithPresentingViewController:(UIViewController *)presentingViewController
+                                       completion:(FUIAuthProviderSignInCompletionBlock)completion;
```
Comment (morganchen12) [inline FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h:96]: @rosalyntan does this approach make sense?

File: FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h [hunk 2]
```diff
@@ -85,6 +89,12 @@ __attribute__((deprecated("Instead use initWithAuthUI:")));
 - (instancetype)initWithScopes:(NSArray <NSString *> *)scopes
 __attribute__((deprecated("Instead use initWithAuthUI:permissions:"))) NS_DESIGNATED_INITIALIZER;
 
+/** @fn initWithAuthUI:scopes:
```
Comment (rosalyntan) [inline FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h:92]: nit: method name here

File: FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m [hunk 3]
```diff
@@ -214,92 +234,77 @@ - (void)signInWithOAuthProvider:(FIROAuthProvider *)oauthProvider
...
+           callback:^(GIDGoogleUser *user, NSError *error) {
+    [self handleSignInWithUser:user
+                         error:error
+      presentingViewController:presentingViewController
+                      callback:^(FIRAuthCredential *credential,
+                                 NSError *error,
+                                 FIRAuthResultCallback result,
+                                 NSDictionary<NSString *,id> *userInfo) {
+      if (completion != nil) {
+        completion(credential, error, result, nil);
```
Comment (rosalyntan) [inline FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m:250]: Should this be `completion(credential, error, result, userInfo)` here?

File: FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m [hunk 4]
```diff
@@ -102,6 +100,13 @@ - (instancetype)initWithScopes:(NSArray *)scopes {
 }
 #pragma clang diagnostic pop
 
+- (GIDSignIn *)googleSignIn {
+  return [GIDSignIn sharedInstance];
```
Comment (rosalyntan) [inline FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m:104]: Should this be `GIDSignIn.sharedInstance`?

From https://developers.google.com/identity/sign-in/ios/quick-migration-guide: "Update all [GIDSignIn sharedInstance] calls to GIDSignIn.sharedInstance property accesses."

File: FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h (hunk 1 above)
Comment (rosalyntan) [reply FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h:96]: I think so -- so will the developer flow for this be to call `signInWithDefaultValue` first to authenticate, and then `requestScopesWithPresentingViewController` to request the appropriate scopes?

Comment (morganchen12) [reply FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h:96]: Yes.

File: FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m (hunk 3 above)
Comment (morganchen12) [reply FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m:250]: Done, good catch!

File: FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m (hunk 4 above)
Comment (morganchen12) [reply FirebaseGoogleAuthUI/Sources/FUIGoogleAuth.m:104]: Done.

File: FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h (hunk 2 above)
Comment (morganchen12) [reply FirebaseGoogleAuthUI/Sources/Public/FirebaseGoogleAuthUI/FUIGoogleAuth.h:92]: Done.

**Review by rosalyntan — APPROVED** [review]

**Pull request merged**